File: zoomx/src/main/java/com/zoomx/zoomx/db/converters/DateConverter.java
package com.zoomx.zoomx.db.converters;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

/**
 * Created by Ahmed Fathallah on 12/6/2017.
 */

public class DateConverter {
    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: zoomx/src/main/java/com/zoomx/zoomx/db/converters/DateConverter.kt
package com.zoomx.zoomx.db.converters

import android.arch.persistence.room.TypeConverter
import java.util.*

/**
 * Created by Ahmed Fathallah on 12/6/2017.
 */

object DateConverter {
    @TypeConverter
    fun toDate(timestamp: Long?): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    @TypeConverter
    fun toTimestamp(date: Date?): Long? {
        return date?.time
    }
}
